### PR TITLE
chore: improve docker tests

### PR DESCRIPTION
**Pull request by divine**
Currently it's not possible to run test when running `docker-compose up` as the test container just stops silently and it's possible to run it again.

After this merge it would be possible to run tests just by starting up the container and it'll just work.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -13,3 +13,9 @@ RUN apt-get update && \
 COPY --from=composer /usr/bin/composer /usr/local/bin/composer
 
 WORKDIR /code
+
+COPY . .
```
Comment (Smolevich) [inline Dockerfile:17]: Better for optimal using docker cache 
```
COPY composer.* ./
RUN composer install
COPY ./ ./
```

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
-ARG PHP_VERSION=8.0
-ARG COMPOSER_VERSION=2.0
+ARG PHP_VERSION=8.1
+ARG COMPOSER_VERSION=2.5.4
 
 FROM composer:${COMPOSER_VERSION}
```
Comment (Smolevich) [inline Dockerfile:4]: Better delete and use in COPY section where binary file is copied

File: docker-compose.yml [hunk 3]
```diff
@@ -3,6 +3,8 @@ version: '3'
 services:
     tests:
         container_name: tests
+        #platform: linux/arm64
```
Comment (Smolevich) [inline docker-compose.yml:6]: Docker-compose cli supports option platform https://github.com/docker/compose/pull/8147

Comment (divine) [reply docker-compose.yml]: That's a good point, thanks!

File: Dockerfile [hunk 4]
```diff
@@ -1,8 +1,5 @@
-ARG PHP_VERSION=8.0
-ARG COMPOSER_VERSION=2.0
-
-FROM composer:${COMPOSER_VERSION}
-FROM php:${PHP_VERSION}-cli
+FROM composer:2.5.4
```
Comment (Smolevich) [inline Dockerfile:1]: Why did you delete `ARG` ?

File: Dockerfile [hunk 5]
```diff
@@ -13,3 +13,9 @@ RUN apt-get update && \
 COPY --from=composer /usr/bin/composer /usr/local/bin/composer
```
Comment (Smolevich) [inline Dockerfile:13]: Better change on 
```
COPY --from=composer:${COMPOSER_VERSION} /usr/bin/composer /usr/local/bin/composer
``` 
And delete first `FROM`

**Comment by Smolevich** [comment]
Add minimal description, please
I understand your motivation but another contributors don't understand reason of your changes

**Comment by divine** [comment]
> Add minimal description, please I understand your motivation but another contributors don't understand reason of your changes

I've added description,

Thank you for your review.

File: Dockerfile (hunk 1 above)
Comment (Smolevich) [reply Dockerfile:16]: @divine can you change ?

Comment (divine) [reply Dockerfile]: @Smolevich done.

**Review by Smolevich — APPROVED** [review]
